### PR TITLE
new stuff

### PR DESCRIPTION
**Pull request by thehorizon84**
* Interaction Menu Freedom
* Weapon Explosion Radius modifier
* Fast Respawn improvements, and added disable wasted sounds to it

### CODE REVIEW

**Comment by gir489returns** [comment]
This entire commit could be a Lua script, why does it need to be in base Yim?

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,7 +40,7 @@ else ()
   add_compile_options(-Wa,-mbig-obj)
 endif ()
 
-add_library(YimMenu MODULE "${SRC_MAIN}")
+add_library(YimMenu MODULE "${SRC_MAIN}" "src/backend/looped/self/interaction_menu_freedom.cpp" "src/backend/looped/weapons/modify_explosion_radius.cpp")
```
Comment (gir489returns) [inline CMakeLists.txt:43]: Why is this here?

Comment (thehorizon84) [reply CMakeLists.txt]: No idea tbh, don't know why that went through

**Comment by thehorizon84** [comment]
> This entire commit could be a Lua script, why does it need to be in base Yim?

Actually, your comments are ignored from now on. Not going to contribute to your drama-cravings.

**Comment by lonelybud** [comment]
Interaction Menu Freedom did  not worked for passive mode.-

Enable it. Then disable it. Then you will see 30 secs wait. Then enable Interaction Menu Freedom and then try to enable again. Then disable it. Then enable it. Meanwhile in the process you will see that when you go back to interaction menu the disable will change to enable vice versa and also the passive mod wont get disabled.

I do not want to test it for any other stuff

**Comment by thehorizon84** [comment]
> Interaction Menu Freedom did  not worked for passive mode.-
> 
> 
> 
> Enable it. Then disable it. Then you will see 30 secs wait. Then enable Interaction Menu Freedom and then try to enable again. Then disable it. Then enable it. Meanwhile in the process you will see that when you go back to interaction menu the disable will change to enable vice versa and also the passive mod wont get disabled.
> 
> 
> 
> I do not want to test it for any other stuff

That's not what it does. Interaction Menu Freedom allows you to access any submenu/section that's greyed out (inaccessible). Enable it and you'll see that you're able to access (for example) Inventory -> Cash, which normally is and has been disabled for many years.

I also believe YimMenu already have a standalone Passive Mode feature?

File: src/backend/looped/self/fast_respawn.cpp [hunk 2]
```diff
@@ -12,14 +11,18 @@ namespace big
...
+
+			// triggers respawn instantly upon death, has no effect if not respawning so no need to check if the player's dead
+			misc::set_bit(&(*scr_globals::freemode_properties.at(1685).at(756).as<int*>()), 1); 
+		}
+
+		virtual void on_disable() override
+		{
+			*scr_globals::disable_wasted_sound.as<bool*>() = false;
+
+			misc::clear_bit(&(*scr_globals::freemode_properties.at(1685).at(756).as<int*>()), 1);
```
Comment (Yimura) [inline src/backend/looped/self/fast_respawn.cpp:25]: Any comments here to find these global offsets in the scripts would be appreciated.

File: src/backend/looped/weapons/modify_explosion_radius.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		Hash ped_weapon;
+
+		virtual void on_tick() override
+		{
+			Hash weapon{};
+			WEAPON::GET_CURRENT_PED_WEAPON(self::ped, &weapon, 0);
+
+			if (weapon != ped_weapon)
+			{
+				WEAPON::SET_WEAPON_AOE_MODIFIER(weapon, 1);
```
Comment (Yimura) [inline src/backend/looped/weapons/modify_explosion_radius.cpp:19]: Is this for all weapons the same AOE?

File: src/core/scr_globals.hpp [hunk 4]
```diff
@@ -58,6 +58,10 @@ namespace big::scr_globals
 
 	static inline const script_global gun_van(1956855);
 
+	static inline const script_global interaction_menu_access(2766623);
+
+	static inline const script_global disable_wasted_sound(2764735);
```
Comment (Yimura) [inline src/core/scr_globals.hpp]: Any comments to find these script globals again?

**Review by Yimura — CHANGES_REQUESTED** [review]
Hey there, for this PR it's fine as the features are small and most likely easy enough to maintain in the future.

For future PR's to YimMenu's main code base we suggest you ask yourself the question, could this be easily made into a Lua script?
You can [open an issue on this repo](https://github.com/YimMenu-Lua/submission/issues/new/choose) to request a Github repository.

Just a few comments to address and then it can be merged into master.
I'm open for suggestions from others if they think this is fine to be added into the main code base or certain parts of this PR could be converted into a Lua script instead.

File: src/backend/looped/weapons/modify_explosion_radius.cpp (hunk 3 above)
Comment (thehorizon84) [reply src/backend/looped/weapons/modify_explosion_radius.cpp:19]: Yes, only affects weapons that create explosions and it works like a multiplier, so 1 is default.

File: src/core/scr_globals.hpp (hunk 4 above)
Comment (thehorizon84) [reply src/core/scr_globals.hpp]: both are very easy to find.

interaction_menu_access (script: am_pi_menu): 
search for PI_BIK_13_2_H and it'll be the global right above that line.

disable_wasted_sound (script: freemode): 
AUDIO::PLAY_SOUND_FRONTEND(-1, "Wasted", "POWER_PLAY_General_Soundset", true);

File: src/backend/looped/self/fast_respawn.cpp (hunk 2 above)
Comment (thehorizon84) [reply src/backend/looped/self/fast_respawn.cpp:25]: Freemode script: search for KILL_STRIP_H and it'll be the first !IS_BIT_SET global above that (it sets to 2)

Comment (Yimura) [reply src/backend/looped/self/fast_respawn.cpp:25]: Could you add it as a comment in the code?

File: src/core/scr_globals.hpp (hunk 4 above)
Comment (Yimura) [reply src/core/scr_globals.hpp]: Could you add it as a comment in the code?

File: src/backend/looped/self/fast_respawn.cpp (hunk 2 above)
Comment (thehorizon84) [reply src/backend/looped/self/fast_respawn.cpp:25]: Absolutely, should be fixed now

**Review by Yimura — APPROVED** [review]
Thanks for adding to comments to make it easier to update in the future.
